templates/child_templates/template.lambda.yaml

```diff
@@ -0,0 +1,217 @@
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - edgelambda.amazonaws.com
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+
+  TheOriginRequestLambdaFunctionRolePolicies:
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyName: "root"
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement: 
+          - 
+            Effect: Allow 
+            Action: "ssm:GetParameter"
+            Resource:
+              - "arn:aws:ssm:*:*:parameter/*"
+      Roles:
+        - Ref: "TheOriginRequestLambdaFunctionExecutionRole"
+
+  TheOriginRequestLambdaFunctionVersion:
+    Type: AWS::Lambda::Version
+    Properties:
+      FunctionName: !Ref TheOriginRequestLambdaFunction
+      Description: !Sub "URL rewriting"
+
+
+  # Handle writing data to SSM param store.
+  ProcessRedirectsLambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: >
+        Lambda function to process _redirects file and populate param store
+      Code:
+        ZipFile: !Sub |
+          var AWS = require('aws-sdk');
+          var s3 = new AWS.S3();
+          var ssm = new AWS.SSM();
+          exports.handler = (event, context, callback) => { 
+            const re = /[^\.](\S+)\b/g;
+            const params = {
+               Bucket: event.Records[0].s3.bucket.name,
+               Key: '_redirects'
+            }
+            let redirectsList;
+            let processedRedirects = []
+            let arr = [];
+            let m =''
+            s3.getObject(params, function(err, data) {
+              if (err) {
+                console.log(err);
+              } else { 
+                redirectsList = data.Body.toString();
+                let entries = redirectsList.split("\n");
+                for (var i in entries) {
+                  while ((m=re.exec(entries[i])) !== null) {
+                    arr.push(m[1]);
+                  }
+                  processedRedirects.push(arr)
+                  arr = []
+                }  
+                let paramsPut = {
+                  Name: 'redirects',
+                  Type: 'String',
+                  Overwrite: true,
+                  Value: JSON.stringify(processedRedirects)
+                }
+                ssm.putParameter(paramsPut, function(err, data) {
+                  if (err) {
+                    console.log(err, err.stack);
+                  } 
+                })
+              } 
+            });  
+          }; 
+      Handler: index.handler
+      MemorySize: 128
+      Role: !Sub ${ProcessRedirectsLambdaFunctionRole.Arn}
+      Runtime: nodejs6.10
+
+  ProcessRedirectsLambdaFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - edgelambda.amazonaws.com
```

Comment:
Will this one get called from the edge? If not, we can remove this line.

Reply:
No, this isn't called from the edge (the S3 bucket calls it via an event). Have removed from the CF template.